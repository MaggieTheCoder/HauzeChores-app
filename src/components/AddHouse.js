/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import propTypes from "prop-types";
import {
  uniqueNamesGenerator,
  adjectives,
  colors,
  animals,
} from "unique-names-generator";
import postNewHouse from "../requests/postNewHouse";
import getHouseByInviteCode from "../requests/getHouseByInviteCode";

const AddHouse = ({ setHouseId, houseId, user }) => {
  const [name, setName] = useState();
  const [visibleItem, setVisibleItem] = useState(false);
  const [inviteCode, setInviteCode] = useState("");
  const [randomStr, setRandomStr] = useState("");

  const handleAddHouse = (event) => {
    setName(event.target.value);
  };

  const handleSubmit = async () => {
    const randomString = uniqueNamesGenerator({
      dictionaries: [adjectives, colors, animals],
    });
    console.log({
      house: name,
      uniqueCode: randomString,
    });
    postNewHouse(name, randomString);
    setInviteCode(randomString);
    setVisibleItem(true);
    setRandomStr(randomString);
  };

  const handleHouseIDChange = () => {
    getHouseByInviteCode(inviteCode, setHouseId, user.email);
  };
  return (
    <div className="add-house">
      {houseId}
      <form onSubmit={handleAddHouse}>
        <input
          className="input-house__name"
          type="text"
          placeholder="type house name"
          onChange={handleAddHouse}
        />
        <button type="button" onClick={() => handleSubmit()}>
          create
        </button>
        {visibleItem && <p>Your unique code is `{randomStr}`</p>}

        <br />
        <input
          className="input-invitation__code"
          type="text"
          placeholder="use invitation code"
        />
        <button type="button">find and join</button>
      </form>
      <button
        type="button"
        onClick={() => {
          handleHouseIDChange();
        }}
      >
        Next
      </button>
    </div>
  );
};

export default AddHouse;

AddHouse.propTypes = {
  setHouseId: propTypes.func.isRequired,
  user: propTypes.shape().isRequired,
  houseId: propTypes.number.isRequired,
};
