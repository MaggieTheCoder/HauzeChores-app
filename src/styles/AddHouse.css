.add-house {
padding-top: 60px;
  padding-bottom: 15px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width : 100%;
  height : 700px;
  background-color: #99E5FF;
  border-radius : 90px;
}
.Addhouse__house {
    width: 50%;
    display: flex;
    flex-direction: column;
    margin-bottom: 30px;
}
.input-house__name {
    font-family: "Menlo", sans-serif;
    font-size: 25px;
    font-style: italic;
    margin-bottom: 15px;
    padding: 5px;
    border-radius: 15px;
    text-align: center;
}
.create_button {
 font-family: "Menlo", sans-serif;
  font-size: 25px;
  margin-bottom: 30px;
  background-color: #C60982;
  border-radius: 15px;
  padding: 5px;
  color: white;
}
.input-invitation__code {
    font-family: "Menlo", sans-serif;
    font-size: 25px;
    font-style: italic;
    margin-bottom: 15px;
    padding: 5px;
    border-radius: 15px;
    text-align: center;
}
.find_button {
    font-family: "Menlo", sans-serif;
    font-size: 25px;
    margin-bottom: 10px;
    background-color: #C60982;
    border-radius: 15px;
    padding: 5px;
    color: white; 
}
.next_button {
    width: 50%;
    display: flex;
    margin-bottom: 30px;
    font-family: "Menlo", sans-serif;
    text-align: center;
    font-size: 25px;
    margin-bottom: 10px;
    background-color: #360866;;
    border-radius: 15px;
    padding: 5px;
    color: white; 
}