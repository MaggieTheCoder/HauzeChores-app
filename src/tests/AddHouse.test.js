import React from "react";
import { render, screen } from "@testing-library/react";
import AddHouse from "../components/AddHouse";

describe("Add House component", () => {
  const validProps = {
    setHouseId: () => {},
    houseId: 2,
    user: {},
    setCode: () => {},
    setUserId: () => {},
  };

  test("It renders correctly", () => {
    const { asFragment } = render(
      <AddHouse
        setHouseId={validProps.setHouseId}
        houseId={validProps.houseId}
        user={validProps.user}
        setCode={validProps.setCode}
        setUserId={validProps.setUserId}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Create house button and input render correctly", () => {
    render(
      <AddHouse
        setHouseId={validProps.setHouseId}
        houseId={validProps.houseId}
        user={validProps.user}
        setCode={validProps.setCode}
        setUserId={validProps.setUserId}
      />
    );
    const creatButton = screen.getByText("CREATE");
    const createInput = screen.getByPlaceholderText("type house name");

    expect(creatButton).toBeInTheDocument();
    expect(createInput).toBeInTheDocument();
  });
  test("Find and join house button and input render correctly", () => {
    render(
      <AddHouse
        setHouseId={validProps.setHouseId}
        houseId={validProps.houseId}
        user={validProps.user}
        setCode={validProps.setCode}
        setUserId={validProps.setUserId}
      />
    );
    const findButton = screen.getByText("FIND AND JOIN");
    const findInput = screen.getByPlaceholderText("use invitation code");

    expect(findButton).toBeInTheDocument();
    expect(findInput).toBeInTheDocument();
  });

  it("Next button renders correctly", () => {
    render(
      <AddHouse
        setHouseId={validProps.setHouseId}
        houseId={validProps.houseId}
        user={validProps.user}
        setCode={validProps.setCode}
        setUserId={validProps.setUserId}
      />
    );
    const nextButton = screen.getByText("NEXT");

    expect(nextButton).toBeInTheDocument();
  });
});
