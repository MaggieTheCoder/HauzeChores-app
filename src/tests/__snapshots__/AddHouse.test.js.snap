// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add House component It renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-house"
  >
    2
    <form
      class="Addhouse__house"
    >
      <input
        class="input-house__name"
        placeholder="type house name"
        type="text"
      />
      <button
        class="create_button"
        type="button"
      >
        CREATE
      </button>
      <br />
      <input
        class="input-invitation__code"
        placeholder="use invitation code"
        type="text"
      />
      <button
        class="find_button"
        type="button"
      >
        FIND AND JOIN
      </button>
    </form>
    <button
      class="next_button"
      type="button"
    >
      NEXT
    </button>
  </div>
</DocumentFragment>
`;
